pom.xml

```diff
@@ -124,6 +126,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>javax.persistence</groupId>
+            <artifactId>javax.persistence-api</artifactId>
```

Comment:
Adding a non optional dependency on the persistence API will solve your requirement, but foist the javax.persistence APIs on everyone else using Jongo.  There must be a better way to rework this, so that the project supports the extension you were looking for in #317 .

Reply:
Yep #317 is the root cause of all evil. But having said that, [MongoJack](https://github.com/mongojack/mongojack) supports persistence API out of the box so why wouldn't Jongo? It's a much better way to annotate a model object.

The other way I can see this working is if you split your `ObjectId` and `MongoObjectId` out into a separate Jongo API package, at least it would be a much smaller api only "dependency" for a model object. Having said that, still believe `javax.persistence.Id` support should be OOTB .

Reply:
With a little refactoring, we could probably get `javax.persistence` to be optional and supported out of the box.  Let me see if I can work up a change that will do that.